### PR TITLE
DOCS: Add glossaries

### PR DESCRIPTION
**Pull request by hiroyuki-sato**
## What

Document improvement. add glossaries

## Why ?

Many acronyms have been used in this project. I think new project members want to know these words' meanings.
I use a reStructured text format. Because It's sort friendly
 

## How ?

Add glossaries page.

glossaries

### CODE REVIEW

**Comment by shamisp** [comment]
ok to test

File: docs/source/glossaries.rst [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+=========== ==============================================================
+Keyword     Description                                                   
+=========== ==============================================================
+ACK         Acknowledge                                                   
+am          Active Message                                                
+AMO         Atomic Memory Operation                                       
+ANL         Argonne National Laboratory                                   
+AZP         AZure Pipeline                                                
+bcopy       Byte copy
```
Comment (shamisp) [inline docs/source/glossaries.rst:13]: Should be buffer copy

File: docs/source/glossaries.rst [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+Keyword     Description                                                   
+=========== ==============================================================
+ACK         Acknowledge                                                   
+am          Active Message                                                
+AMO         Atomic Memory Operation                                       
+ANL         Argonne National Laboratory                                   
+AZP         AZure Pipeline                                                
+bcopy       Byte copy                                                     
+Bistro      Binary Instrumentation                                        
+BTL         Byte Transfer Layer
```
Comment (shamisp) [inline docs/source/glossaries.rst:15]: Open MPI's Byte Transfer Layer

File: docs/source/glossaries.rst [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+ACK         Acknowledge                                                   
+am          Active Message                                                
+AMO         Atomic Memory Operation                                       
+ANL         Argonne National Laboratory                                   
+AZP         AZure Pipeline                                                
+bcopy       Byte copy                                                     
+Bistro      Binary Instrumentation                                        
+BTL         Byte Transfer Layer                                           
+cm          Connection Manager                                            
+CMA         Cross Memory Attach
```
Comment (shamisp) [inline docs/source/glossaries.rst:17]: Linux Kernel Cross Memory Attach

File: docs/source/glossaries.rst [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+bcopy       Byte copy                                                     
+Bistro      Binary Instrumentation                                        
+BTL         Byte Transfer Layer                                           
+cm          Connection Manager                                            
+CMA         Cross Memory Attach                                           
+CQ          Completion Queue(Infiniband)                                  
+CQE         Completion Queue Entry(Infiniband)                            
+csmock      static analysis tools                                         
+CUDA        Compute Unified Device Architecture(NVIDIA)                   
+DC          Dynamically Connected transport(Infiniband)
```
Comment (shamisp) [inline docs/source/glossaries.rst:22]: Dynamically Connected transport(Mellanox/Nvidia Infiniband)  
(DC is not part of IBTA spec)

File: docs/source/glossaries.rst [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+am          Active Message                                                
+AMO         Atomic Memory Operation                                       
+ANL         Argonne National Laboratory                                   
+AZP         AZure Pipeline                                                
+bcopy       Byte copy                                                     
+Bistro      Binary Instrumentation                                        
+BTL         Byte Transfer Layer                                           
+cm          Connection Manager                                            
+CMA         Cross Memory Attach                                           
+CQ          Completion Queue(Infiniband)
```
Comment (shamisp) [inline docs/source/glossaries.rst:18]: Infiniband  -> InfiniBand

File: docs/source/glossaries.rst [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+GDR         GPUDirect RDMA                                                
+gtest       Google Test                                                   
+HPC         High Performance Computing                                    
+HWTM        HardWare Tag Matching                                         
+IB          Infiniband                                                    
+iface       Interaface                                                    
+IPC         Inter Process Communication                                   
+JUCX        Java API over UCP                                             
+KLM         A new sophisticated way of creating memory regions.(Mellanox) 
+KNEM        Kernel Nemesis
```
Comment (shamisp) [inline docs/source/glossaries.rst:36]: Inria's Kernel module for process to process zero copy (https://knem.gitlabpages.inria.fr)

File: docs/source/glossaries.rst [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+JUCX        Java API over UCP                                             
+KLM         A new sophisticated way of creating memory regions.(Mellanox) 
+KNEM        Kernel Nemesis                                                
+LDPM        Legacy DPM                                                    
+LLNL        Lawrence Livermore National Laboratory                        
+madvise     give advice about use of memory. See manual madvise(2)        
+md          Memory Domain                                                 
+MEMH        Memory Handle                                                 
+MLX         Mellanox Technologies                                         
+mlx5        Connect-X5 VPI
```
Comment (shamisp) [inline docs/source/glossaries.rst:43]: Mellanox-5 generation of Mellanox device driver (I think all connectX-4 generation and above)

File: docs/source/glossaries.rst [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+LLNL        Lawrence Livermore National Laboratory                        
+madvise     give advice about use of memory. See manual madvise(2)        
+md          Memory Domain                                                 
+MEMH        Memory Handle                                                 
+MLX         Mellanox Technologies                                         
+mlx5        Connect-X5 VPI                                                
+mm          Memory Mapper                                                 
+MP-XRQ      Multi-Packet XRQ                                              
+MPI         Message Passing INterface                                     
+MPICH       A MPI Implementation
```
Comment (shamisp) [inline docs/source/glossaries.rst:47]: ANL MPI Implementation

File: docs/source/glossaries.rst [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+UCX         Unified Communication X                                       
+UD          Unreliable Datagram (Infiniband)                              
+uGNI        user level generic network interface(Cray)                    
+UMR         User mode memory registration                                 
+VPI         Virtual Protocol Interconnect                                 
+WIP-DNM     Work-in-Progress-Do-Not-Merge                                 
+WQ          Work Queue(Infiniband)                                        
+WQE         Work Queue Elements (pronounce WOOKIE)                        
+WR          Work Request                                                  
+XPMEM       cross partition memory
```
Comment (shamisp) [inline docs/source/glossaries.rst:102]: Linux kernel module originally developed by SGI implementing cross process memory mapping with  zero copy (https://github.com/hjelmn/xpmem)

File: docs/source/glossaries.rst (hunk 4 above)
Comment (hiroyuki-sato) [reply docs/source/glossaries.rst]: done. modified Infiniband -> InfiniBand.

File: docs/source/glossaries.rst (hunk 5 above)
Comment (hiroyuki-sato) [reply docs/source/glossaries.rst]: Changed all of Infiniband -> InfiniBand

File: docs/source/glossaries.rst (hunk 6 above)
Comment (hiroyuki-sato) [reply docs/source/glossaries.rst]: done

File: docs/source/glossaries.rst (hunk 9 above)
Comment (hiroyuki-sato) [reply docs/source/glossaries.rst]: I modified this. But, I think it's a little longer description.

**Comment by hiroyuki-sato** [comment]
@shamisp Thank you for your comment! I fixed it.
I removed trailing spaces. So I attached diff without comparing white space.

```diff
diff --git a/docs/source/glossaries.rst b/docs/source/glossaries.rst
index 2acdc6c3c..be8dc1272 100644
--- a/docs/source/glossaries.rst
+++ b/docs/source/glossaries.rst
@@ -2,24 +2,24 @@
 Glossaries
 **********
 
-=========== ==============================================================
+=========== ====================================================================================================================================================
 Keyword     Description
-=========== ==============================================================
+=========== ====================================================================================================================================================
 ACK         Acknowledge
 am          Active Message
 AMO         Atomic Memory Operation
 ANL         Argonne National Laboratory
 AZP         AZure Pipeline
-bcopy       Byte copy                                                     
+bcopy       Buffered copy
 Bistro      Binary Instrumentation
-BTL         Byte Transfer Layer                                           
+BTL         Open MPI's Byte Transfer Layer
 cm          Connection Manager
-CMA         Cross Memory Attach                                           
-CQ          Completion Queue(Infiniband)                                  
-CQE         Completion Queue Entry(Infiniband)                            
+CMA         Linux Kernel Cross Memory Attach
+CQ          Completion Queue(InfiniBand)
+CQE         Completion Queue Entry(InfiniBand)
 csmock      static analysis tools
 CUDA        Compute Unified Device Architecture(NVIDIA)
-DC          Dynamically Connected transport(Infiniband)                   
+DC          Dynamically Connected transport(Mellanox/Nvidia InfiniBand)
 DPM         Direct Packet Mode
 ep          EndPoint
 FC          Flow Control
@@ -28,23 +28,23 @@ GDR         GPUDirect RDMA
 gtest       Google Test
 HPC         High Performance Computing
 HWTM        HardWare Tag Matching
-IB          Infiniband                                                    
+IB          InfiniBand
 iface       Interaface
 IPC         Inter Process Communication
 JUCX        Java API over UCP
 KLM         A new sophisticated way of creating memory regions.(Mellanox)
-KNEM        Kernel Nemesis                                                
+KNEM        Inria's Kernel module for process to process zero copy (https://knem.gitlabpages.inria.fr)
 LDPM        Legacy DPM
 LLNL        Lawrence Livermore National Laboratory
 madvise     give advice about use of memory. See manual madvise(2)
 md          Memory Domain
 MEMH        Memory Handle
 MLX         Mellanox Technologies
-mlx5        Connect-X5 VPI                                                
+mlx5        Mellanox-5 generation of Mellanox device driver
 mm          Memory Mapper
 MP-XRQ      Multi-Packet XRQ
 MPI         Message Passing INterface
-MPICH       A MPI Implementation                                          
+MPICH       ANL MPI Implementation
 MTT         The MPI Testing Tool
 NAK         Negative Acknowledge
 ODP         OnDemand Paging
@@ -61,8 +61,8 @@ PGAS        Partitioned Global Address Space
 POSIX       Portable operating system interface
 ppn         processes per node
 PR          Pull Request
-QP          Queue Pair(Infiniband)                                        
-RC          Reliable Connection (Infiniband)                              
+QP          Queue Pair(InfiniBand)
+RC          Reliable Connection (InfiniBand)
 rcache      Registration Cache
 RDMA        Remote Direct Memory Access
 REQ         Request
@@ -75,7 +75,7 @@ RTE         Run Time Environment
 RX          Receive
 Skb         Socket Buffer
 sm          Shared Memory
-SM          Subnet Manager(Infiniband)                                    
+SM          Subnet Manager(InfiniBand)
 SockCM      Socket Connection Manager
 SRQ         Shared Receive Queue
 SYSV        UNIX System V
@@ -83,7 +83,7 @@ tl          Transport Layer
 TLS         Transpot LayerS
 TM          Tag Matching
 TX          Transmit
-UC          Unreliable Connection (Infiniband)                            
+UC          Unreliable Connection (InfiniBand)
 UCC         Unified Collective Communications
 UCF         Unified Communication Framework
 UCM         Unified Communication Memory
@@ -91,17 +91,17 @@ UCP         Unified Communication Protocols Higher level API
 UCS         Unified Communication Service Common utilities.
 UCT         Unified Communication Transport Lower level API
 UCX         Unified Communication X
-UD          Unreliable Datagram (Infiniband)                              
+UD          Unreliable Datagram (InfiniBand)
 uGNI        user level generic network interface(Cray)
 UMR         User mode memory registration
 VPI         Virtual Protocol Interconnect
 WIP-DNM     Work-in-Progress-Do-Not-Merge
-WQ          Work Queue(Infiniband)                                        
+WQ          Work Queue(InfiniBand)
 WQE         Work Queue Elements (pronounce WOOKIE)
 WR          Work Request
-XPMEM       cross partition memory                                        
-XRC         eXtended Reliable Connection(Infiniband)                      
+XPMEM       `XPMEM <https://github.com/hjelmn/xpmem>`_ Linux kernel module originally developed by SGI implementing cross process memory mapping with zero copy
+XRC         eXtended Reliable Connection(InfiniBand)
 XRCD        XRC Domains
 XRQ         Mellanox Shared Receive Queue
 Zcopy       Zero Copy
-=========== ==============================================================
+=========== ====================================================================================================================================================
```

**Comment by hiroyuki-sato** [comment]
Could someone rerun CI? This PR modify document only. So `gtest` will should complete.

File: docs/source/glossaries.rst [hunk 10]
```diff
@@ -0,0 +1,107 @@
+**********
+Glossaries
+**********
+
+=========== ====================================================================================================================================================
+Keyword     Description
+=========== ====================================================================================================================================================
+ACK         Acknowledge
+am          Active Message
```
Comment (yosefe) [inline docs/source/glossaries.rst:9]: AM (pls use upper case for all acronyms)

File: docs/source/glossaries.rst [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+AMO         Atomic Memory Operation
+ANL         Argonne National Laboratory
+AZP         AZure Pipeline
```
Comment (yosefe) [inline docs/source/glossaries.rst:12]: Azure

File: docs/source/glossaries.rst [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+fd          File Descriptor
+GDR         GPUDirect RDMA
+gtest       Google Test
+HPC         High Performance Computing
+HWTM        HardWare Tag Matching
+IB          InfiniBand
+iface       Interaface
+IPC         Inter Process Communication
+JUCX        Java API over UCP
+KLM         A new sophisticated way of creating memory regions.(Mellanox)
```
Comment (yosefe) [inline docs/source/glossaries.rst:35]: remove `.`

File: docs/source/glossaries.rst [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+KNEM        Inria's Kernel module for process to process zero copy (https://knem.gitlabpages.inria.fr)
+LDPM        Legacy DPM
+LLNL        Lawrence Livermore National Laboratory
+madvise     give advice about use of memory. See manual madvise(2)
```
Comment (yosefe) [inline docs/source/glossaries.rst:39]: ` See madvise (2)`

File: docs/source/glossaries.rst [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+mm          Memory Mapper
+MP-XRQ      Multi-Packet XRQ
+MPI         Message Passing INterface
+MPICH       ANL MPI Implementation
+MTT         The MPI Testing Tool
+NAK         Negative Acknowledge
+ODP         OnDemand Paging
+OFA         OpenFabrics Alliance
+OMPI        OpenMPI
+OOB         Out of band
```
Comment (yosefe) [inline docs/source/glossaries.rst:53]: Out of band / Out of box

File: docs/source/glossaries.rst [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+OOO         Out of Order
+OPA         Omni-Path Architecture
+Open MPI    A MPI Implementation
+ORNL        Oak Ridge National Laboratory
+PCIe        PCI Express
+PFN         Physical Page Frame
```
Comment (yosefe) [inline docs/source/glossaries.rst:59]: Physical Frame Number

File: docs/source/glossaries.rst [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+ppn         processes per node
+PR          Pull Request
+QP          Queue Pair(InfiniBand)
+RC          Reliable Connection (InfiniBand)
+rcache      Registration Cache
+RDMA        Remote Direct Memory Access
+REQ         Request
+rkey        Remote KEY
+RMA         Remote Memory Access
+RNR         Receiver Not Ready
```
Comment (yosefe) [inline docs/source/glossaries.rst:71]: RnR

File: docs/source/glossaries.rst [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+RoCE        RDMA over Converged Ethernet
+ROCm        Radeon Open Compute platform(AMD)
+RTE         Run Time Environment
+RX          Receive
+Skb         Socket Buffer
```
Comment (yosefe) [inline docs/source/glossaries.rst:76]: skb

File: docs/source/glossaries.rst [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+sm          Shared Memory
+SM          Subnet Manager(InfiniBand)
+SockCM      Socket Connection Manager
+SRQ         Shared Receive Queue
+SYSV        UNIX System V
```
Comment (yosefe) [inline docs/source/glossaries.rst:81]: SysV

File: docs/source/glossaries.rst [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
+tl          Transport Layer
+TLS         Transpot LayerS
+TM          Tag Matching
+TX          Transmit
+UC          Unreliable Connection (InfiniBand)
+UCC         Unified Collective Communications
+UCF         Unified Communication Framework
+UCM         Unified Communication Memory
+UCP         Unified Communication Protocols Higher level API
+UCS         Unified Communication Service Common utilities.
```
Comment (yosefe) [inline docs/source/glossaries.rst:91]: remove `.`

File: docs/source/glossaries.rst (hunk 10 above)
Comment (hiroyuki-sato) [reply docs/source/glossaries.rst]: done. I modified some words to uppercase. Please take a look again.

**Comment by hiroyuki-sato** [comment]
Hello, @yosefe Thank you for your review. Please take a look again when you get a chance.

**Comment by hiroyuki-sato** [comment]
Thank you!

**Pull request merged**